src/js/components/FormGroupComponent.jsx

```diff
@@ -2,17 +2,31 @@ var React = require("react/addons");
 
 var FormGroupComponent = React.createClass({
   displayName: "FormGroupComponent",
+
   propTypes: {
     attribute: React.PropTypes.string,
     children: React.PropTypes.object.isRequired,
     errors: React.PropTypes.array,
     help: React.PropTypes.string,
     label: React.PropTypes.string,
-    model: React.PropTypes.object.isRequired
+    model: React.PropTypes.object.isRequired,
+    validator: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      validationError: null
+    };
   },
 
   onInputChange: function (event) {
-    this.props.model.set(event.target.name, event.target.value);
+    var props = this.props;
+
+    props.model[event.target.name] = event.target.value;
```

Comment:
I agree, this is very nasty.

How should this work with the internal input state? How do we get the modified value back into the object in NewAppModalComponent?
Should we define an own setter-function on the object?
Then we could do this again:
```
this.props.model.set(event.target.name, event.target.value);
```